src/sqlfluff/core/parser/segments/base.py

```diff
@@ -1219,79 +1219,90 @@ def apply_fixes(
         after = []
         fixes_applied: List[LintFix] = []
         todo_buffer = list(self.segments)
-        while True:
-            if len(todo_buffer) == 0:
-                break
-            else:
-                seg = todo_buffer.pop(0)
+        requires_validate = False
+
+        while todo_buffer:
+            seg = todo_buffer.pop(0)
 
-                # Look for uuid match.
-                # This handles potential positioning ambiguity.
-                anchor_info: Optional[AnchorEditInfo] = fixes.pop(seg.uuid, None)
-                if anchor_info is not None:
-                    seg_fixes = anchor_info.fixes
+            # Look for uuid match.
+            # This handles potential positioning ambiguity.
+            anchor_info: Optional[AnchorEditInfo] = fixes.pop(seg.uuid, None)
+
+            if anchor_info is None:
```

Comment:
Flip this `if` to deal with the `is None` case first and then `continue`, followed by an un nested opposite case.